src/strtree.c

```diff
@@ -129,6 +196,54 @@ void query_callback(void *item, void *user_data)
     kv_push(npy_intp, *(npy_intp_vec *)user_data, (npy_intp) item);
 }
 
+/* Evaluate the predicate function against a prepared version of geom
+ * for each geometry in the tree specified by indexes in target_indexes.
+ * out_indexes is updated in place with the indexes of the geometries in the
+ * tree that meet the predicate.
+ * Returns the number of geometries that met the predicate.
+ * Returns -1 in case of error.
+ * */
+
+static int evaluate_predicate(FuncGEOS_YpY_b *predicate_func,
+                                        GEOSGeometry *geom,
+                                        PyArrayObject * tree_geometries,
+                                        npy_intp_vec *in_indexes,
+                                        npy_intp_vec *out_indexes)
+{
+    GEOSContextHandle_t context = geos_context[0];
+    GEOSGeometry *target_geom;
+    const GEOSPreparedGeometry *pgeom;
+    npy_intp i, size, index;
+    int count = 0;
+    npy_intp *geom_ptr;
+    PyArrayObject *result;
+
+    // Create prepared geometry
+    pgeom = GEOSPrepare_r(context, geom);
+    if (pgeom == NULL) {
+        return -1;
+    }
+
+    size = kv_size(*in_indexes);
+    for (i = 0; i < size; i++) {
+        // get index for right geometries from in_indexes
+        index = kv_A(*in_indexes, i);
+
+        // get GEOS geometry from pygeos geometry at index in tree geometries
+        geom_ptr = PyArray_GETPTR1(tree_geometries, index);
+        get_geom(*(GeometryObject **) geom_ptr, &target_geom);
```

Comment:
After https://github.com/pygeos/pygeos/pull/100, we can probably avoid needing `get_geom` here